docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,160 @@
+---
+
+Apollo supports two separate but related features called **persisted queries** and **automated persisted queries** (APQs).
+With both features, clients can execute a GraphQL operation by sending an operation's ID instead of the entire string. An operation's ID is the SHA256 hash of the full operation string. Querying by ID can significantly reduce latency and bandwidth usage for very large operation strings.
+
+## Differences between persisted queries and APQ
+
+The persisted queries feature requires operations to be preregistered in a **persisted query list** (**PQL**). 
+This allows the PQL to act as a safelist of trusted operations made by your first-party apps. As such, persisted queries is a security feature as much as a performance one.
+
+With APQs, if the server can't find the operation ID the client provides, the server returns an error indicating that it needs the full operation string. If the Apollo iOS client receives this error, it automatically retries the operation with the full operation string.
+
+If you _only_ want to improve request latency and bandwidth usage, APQ addresses your use case. If you _also_ want to secure your supergraph with operation safelisting, you should preregister trusted operations in a PQL.
+
+For more details on differences between persisted queries and APQ, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries#differences-from-automatic-persisted).
+
+### Requirements
+
+Persisted queries is currently in [preview](/resources/product-launch-stages#preview) and has the following requirements:
+- Apollo iOS (v1.3.1+)
+- [Apollo Router](/router) (v1.25.0+)
+- [GraphOS Enterprise plan](/graphos/enterprise/)
+
+You can use APQ with the following versions of Apollo iOS, Apollo Router, and Apollo Server:
+- Apollo iOS (v1.0.0+)
+- [Apollo Router](/router) (v0.1.0+)
+- [Apollo Server](/apollo-server/) (v1.0.0+)
+
+## Implementation steps
+
+Both persisted queries and APQ require you to configure code generation and how your client makes requests. The rest of this article details these configurations.
+
+Persisted queries also requires you to create and link a PQL and configure your router. For more information on the other configuration aspects of persisted queries, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries).
+
+### 1. Configure code generation
+
+The code generation requirements for persisted queries and APQ are the same: your operations need a SHA256 hash identifier included in the generated operation objects. To configure code generation accordingly, make the following changes in your `ApolloCodegenConfiguration` JSON file:
+
+1. In the `output` section of your configuration, set the [`operationManifest`](./../code-generation/codegen-configuration#operation-manifest) with:
+- a `path` to output the manifest file
+- the `version`: either `persistedQueries` for persisted queries or `legacyAPQs` for APQs
```

Comment:
Before making this feature stable, should we be naming this `legacyAPQs` or just `APQs`? I was under the assumption that once `persistedQueries` was public availability, that the `APQ` feature would be deprecated across the entire platform. But is that assumption correct? @glasser

Reply:
I like `APQ`! They are very easy to setup, we use them in [Confetti](https://github.com/joreilly/Confetti). I would be sad to see them go but I'll follow what's decided here.

Reply:
@glasser what's your read on this?

Reply:
My read is that APQs are still a thing and not going anywhere. The "old apollo cli" manifest format on the other hand can be deprecated. I would link the "old apollo cli" manifest format to APQ. 

In Apollo Kotlin we called that format `operationOutput` (I think I can remember the apollo cli using something like that but I wouldn't bet on it). But any format like `legacyManifest` would do.

Reply:
So does that mean that the new manifest format can still be used to persist queries that work with APQs? Has apollo server been updated to support this?

Reply:
To be clear, I understand that APQs _generally_ don't need the manifest, since they are automatically-persisted. I'm talking specifically about users that have been using the existing APQ functionality to work as a safelisting mechanism by pre-registering the operations with a manifest.

Reply:
I don't know much about the uses of the older manifest format here — was it for the legacy operation registry/safelisting feature (which was not connected in any way to APQs)? We will certainly be eventually shutting that feature off (it has been undocumented and barred to new users for a while now).

We're not intending to deprecate APQs, which are an easy to set up performance feature (but don't provide any security benefits).

Reply:
Okay, so I was under the impression that the legacy safelisting feature was connected to APQs. If it will be eventually shut off, then we can rename this to just `legacy` and move forward. Thanks @glasser